### DISCUSSION TITLE
Updating index.html from public folder

### DISCUSSION THREAD

**Discussion by therealjuan** [vercel/community#2211] [Help] [answered]
### Summary

Hello,

I have a node.js server with three different routes with:
/ – This loads contents from the public/index.html
/update – serverless function that updates a json file stored in S3
/generate – serverless function that generates a HTML file from the JSON file stored in S3 

I am looking into ways to update the public/index.html when triggering the /generate serverless function. What would be the recommendation for this?

Thanks in advance

**Comment by maddsua** [comment, marked as answer]
I don't think you can modify static assets, HTML files and so on, that had been already deployed. You may try adding some JS logic to the index.html, so it would modify DOM after the original content is loaded. The idea is the same as for vanilla JS, and for any JS framework too: load page -> fetch some data (so we know that something was changed) -> modify contents. Or make the `/` an Edge function, and serve already modified page.
Idk what's your level of expertise is, I'd like to help if more context is provided
